types/react-beautiful-dnd/tslint.json

```diff
@@ -1,3 +1,6 @@
 {
-    "extends": "dtslint/dt.json"
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "strict-export-declare-modifiers": false
```

Comment:
I don't think we should be changing the TSLint defaults. Should you instead update the definitions to adhere to `strict-export-declare-modifiers`?

Reply:
Here is my logic for that. Please, correct me if you see it differently.

I am including `Props` types for the 3 exposed React.Component classes, along with their dependencies. But they are not meant to be used by users, which is illustrated by not exposing them [by the library itself](https://github.com/atlassian/react-beautiful-dnd/blob/master/src/index.js). I want to keep them not exposed to the user to reduce the chance for confusion.

A possible option is to inline all the prop types but that goes agains my other goal - I was trying to stick to the structure of types as defined in original source code for easier maintainability.

I tried to disable this rule in the relevant part of the code, but that is disabled by the custom tslint setup.

Reply:
Interesting — thanks for the detailed explanation! I can't find any better solution than what you’ve done. I still don’t completely understand the rules around whether it's valid for exported interface to reference a non-exported interface. It appears to work in the `.d.ts`, but I remember that doing the same in a `.ts` and having `tsc --declaration` complain.

Reply:
Not exporting the props is going to cause issues for consumers who wrap the components which use them with higher order components. Such components are usually written as a function or ctor with a signature like `wrap<TProp, TInstance extends JSX.Element>(ctor: new (props: TProp): TInstance): MyComponentWrap<TInstance, TProp>;`. You could then have a `export const MyWrappedDraggable = wrap(Draggable)`, which will then need to be printed in declaration emit as `export const MyWrappedDraggable: MyComponentWrap<Draggable, DraggableInternalProps>`. If `DraggableInternalProps` is not visible (because it's not exported it can never be visible outside this module), then users will run into issues where the type can't be named, and they'll need to explicitly annotate `export const MyWrappedDraggable: MyComponentWrap<Draggable, { /*some props assignable to DraggableInternalProps*/ }> = wrap(Draggable)`. Not the greatest experience. That's why you should export all the interfaces you use.

Reply:
Thanks, will do!